app/models/user.rb

```diff
@@ -18,14 +19,30 @@ def remember
     update_attribute(:remember_digest, User.digest(remember_token))
   end
 
-  def authenticated? remember_token
-    BCrypt::Password.new(remember_digest).is_password?(remember_token)
+  def authenticated? attribute, token
+    digest = send("#{attribute}_digest")
+    return false if digest.nil?
+    BCrypt::Password.new(digest).is_password?(token)
   end
 
   def forget
     update_attribute(:remember_digest, nil)
   end
 
+  def create_activation_digest
+    self.activation_token = User.new_token
+    self.activation_digest = User.digest(activation_token)
+  end
+
+  def activate
+    update_attribute(:activated, true)
+    update_attribute(:activated_at, Time.zone.now)
```

Comment:
Sao ko dùng update_attributes để update multi column anh nhỉ

Reply:
update_attributes(:activated => true, :activated_at => Time.zone.now)
this right or wrong???